test/ExtractGoal.lean

```diff
@@ -52,3 +52,26 @@ example {z : Int} : ∃ n : Nat, ↑n = z := by
   set_option pp.all true in
   extract_goal
   sorry
+
+-- Contradiction proof gives full context:
+/--
+info: theorem extracted_1 (h : 1 = 2) : False := sorry
+---
+warning: declaration uses 'sorry'
+-/
+#guard_msgs in
+example (h : 1 = 2) : False := by
+  extract_goal
+  sorry
```

Comment:
```suggestion
/--
info: theorem extracted_1 (h : 1 = 2) : False := sorry
-/
#guard_msgs in
example (h : 1 = 2) : False := by
  extract_goal
  cases h
```
To avoid a `sorry`!

Reply:
I'd figured that if you're doing `extract_goal` you might not know how to prove it yet :smile: I'll mark this as resolved, but if you'd prefer `cases h` feel free to unresolve.